### PR TITLE
Tensor API changes and new commands

### PR DESCRIPTION
**Pull request by hhsecond**
- Tensor API now accept and return numpy arrays by default
- Added `loadbackend`
- Added `scriptdel` and `modeldel`
- mypy annotations, the python3.x way
- Code clean up

### CODE REVIEW

File: redisai/utils.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    elif isinstance(s, bytes):
+        return s.decode('utf-8')
+    else:
+        return s  # Not a string we care about
+
+
+def str_or_strlist(v):
+    if isinstance(v, six.string_types):
+        return [v]
+    return v
```
Comment (lantiga) [inline redisai/utils.py]: Nit: we don't ensure that `v` is a list if it's not a string, so the function name is misleading

**Comment by hhsecond** [comment]
@lantiga There are few more changes I have in mind which removes the `Tensor` or `BlobTensor` class completely. There is no reason to have those classes hanging around if `tensorset` could take list and numpy arrays. I'll change those sometime this week and then there were few folks who offered me their help in rewriting the example repo to handle this change.

What is your ETA for https://github.com/RedisAI/RedisAI/pull/186. I can wait till that lands to finish off this PR and we'll release 0.4 with all these?

**Comment by hhsecond** [comment]
Depens on https://github.com/RedisAI/RedisAI/pull/188

**Comment by hhsecond** [comment]
@lantiga Do you mind another look? We can do another quick release with `deviceid` and `cleaning up of Tensor APIs` if that's OK with you

**Comment by hhsecond** [comment]
Thank you Luca

**Pull request merged**